beetsplug/embedart.py

```diff
@@ -86,12 +86,23 @@ def embed_func(lib, opts, args):
                                     help='extract an image from file metadata')
         extract_cmd.parser.add_option('-o', dest='outpath',
                                       help='image output file')
+        extract_cmd.parser.add_option('-a', dest='albums', action='store_true',
+                                      help='extract the art of all matching '
+                                           'albums')
 
         def extract_func(lib, opts, args):
-            outpath = normpath(opts.outpath or config['art_filename'].get())
-            for item in lib.items(decargs(args)):
-                if self.extract(outpath, item):
+            outpath = opts.outpath or config['art_filename'].get()
+            if opts.albums:
+                if opts.outpath and '/' in opts.outpath.replace('\\', '/'):
```

Comment:
Ah, cool, thanks for clarifying. How about just checking whether `os.path.dirname(outpath)` is non-empty?